apps/passport-client/components/shared/PCDArgs.tsx

```diff
@@ -147,20 +148,36 @@ export function ArgInput<T extends PCDPackage, ArgName extends string>({
   );
 
   if (isStringArgument(arg)) {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
     return <StringArgInput {...props} />;
   } else if (isNumberArgument(arg)) {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
     return <NumberArgInput {...props} />;
   } else if (isBigIntArgument(arg)) {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
     return <BigIntArgInput {...props} />;
   } else if (isBooleanArgument(arg)) {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
     return <BooleanArgInput {...props} />;
   } else if (isToggleListArgument(arg)) {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
     return <ToggleListArgInput {...props} />;
   } else if (isObjectArgument(arg)) {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
     return <ObjectArgInput {...props} />;
   } else if (isPCDArgument(arg)) {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
     return <PCDArgInput {...props} />;
   } else if (isStringArrayArgument(arg)) {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
```

Comment:
rather than `ts-ignore` i suggest casting `as unknown as <whatever the required proptype is>` if possible

Reply:
The problem here is that the `isValid` function in the `props` has the wrong type, and I can't cast it here:

```
Type '{ arg: ArgsOf<T>[ArgName]; argName: string; setArg: (value: ArgsOf<T>[ArgName]["value"]) => void; isValid: (arg: RawValueType<ArgsOf<T>[ArgName]>) => boolean; }' is not assignable to type 'ArgInputProps<StringArgument>'.
  Types of property 'isValid' are incompatible.
    Type '(arg: RawValueType<ArgsOf<T>[ArgName]>) => boolean' is not assignable to type '(arg: string) => boolean'.
      Types of parameters 'arg' and 'arg' are incompatible.
        Type 'string' is not assignable to type 'RawValueType<ArgsOf<T>[ArgName]>'.ts(2322)
function StringArgInput({ arg, setArg, ...rest }: ArgInputProps<StringArgument>): JSX.Element
```